Dockerfile.tool-router.optimized

```
@@ -0,0 +1,61 @@
+# Copy and build application
+COPY tool_router /app/tool_router
+RUN python -m compileall tool_router && \
+    find /usr/local/lib/python3.12/site-packages -name "*.pyc" -delete && \
+    find /usr/local/lib/python3.12/site-packages -name "__pycache__" -type d -exec rm -rf {} + 2>/dev/null || true
+
+# Stage 3: Minimal runtime image
+FROM python:3.12-alpine AS runtime
+WORKDIR /app
+
+# Install only runtime dependencies
+RUN apk add --no-cache curl && \
```

Hadolint (2.14.0)

[warning] Pin versions in apk add. Instead of `apk add <package>` use `apk add <package>=<version>`

(DL3018)